content/blog/automated-releases.md

```diff
@@ -0,0 +1,346 @@
+
+- Tag Git repository with a new version
+- Create GitHub Release
+- Push new version of the package to [NPM](https://www.npmjs.com/)
+- Push new version of Docker image to [Docker Hub](https://hub.docker.com/)
+- Bump the version of the package in `package.json` file and commit the change to the repository
+
+This is how the design looks like:
+
+![npm docker release workflow](/images/posts/release-workflow.png)
+
+There are two workflows designed here. 
+
+The first workflow reacts to changes in the release branch (`master` in this case) and decides if release should be triggered, and triggers it. The last step of the workflow is a pull request creation with changes in `package.json`. Why are changes not committed directly to the release branch? Because in the majority of cases, people, like we in AsyncAPI, use branch protection rules and do not allow direct commits to release branches.
+
+The second workflow is just for handling changes in `package.json`. To fulfill branch protection settings, we had to auto-approve the pull request so we can automatically merge it.
+
+You can extend this workflow with additional steps, like:
+
+* Integration testing
+* Deployment
+* Notifications
+
+## GitHub Actions
+
+Even though I have my opinion about GitHub Actions (for more details, you can read my [post about it](https://dev.to/derberg/github-actions-when-fascination-turns-into-disappointment-4d75)) I still think it is worth investing in it, especially for the release workflows.
+
+We used the GitHub-provided actions and the following awesome actions provided by the community:
+
+- [Create Pull Request](ttps://github.com/marketplace/actions/create-pull-request)
+- [Auto Approve](https://github.com/marketplace/actions/auto-approve)
+- [Merge Pull Request](https://github.com/marketplace/actions/merge-pull-requests)
+
+### Release workflow
+
+Release workflow triggers every time there is something new happening in the release branch. In our case, it is a `master` branch:
+
+```yaml
+on:
+  push:
+    branches:
+      - master
+```
+
+#### GitHub and NPM
+
+For releases to GitHub and NPM, the most convenient solution is to integrate [semantic release](https://github.com/semantic-release/semantic-release) package and related plugins that support Conventional Commits. You can configure plugins in your `package.json` in the order they should be invoked:
+
+```json
+"plugins": [
+  [
+    "@semantic-release/commit-analyzer",
+    {
+      "preset": "conventionalcommits"
+    }
+  ],
+  [
+    "@semantic-release/release-notes-generator",
+    {
+      "preset": "conventionalcommits"
+    }
+  ],
+  "@semantic-release/npm",
+  "@semantic-release/github"
+]
+```
+
+Remember that functional automation should use a technical bot rather than a real user. GitHub Actions allow you to encrypt credentials to different systems on the repository level and referring them in actions looks as follows: 
```

Comment:
```suggestion
Conveniently, functional automation uses a technical bot rather than a real user. GitHub actions allow you to encrypt the credentials of different systems at the repository level. Referring to them in actions looks as follows: 
```

Regarding replacing "remember" with "conveniently": It would be great if you could point to some online resources explaining the reasons. 

Github Actions -> Singular, name of the product.
Github actions -> Plural, the actions of your workflow.

Also, I split the last sentence to make it shorter and easier to read.